torchrl/envs/transforms/rlhf.py

```diff
@@ -4,18 +4,413 @@
 import torch
-from tensordict import TensorDict, TensorDictBase, unravel_key
+from tensordict import (
+    maybe_dense_stack,
+    NestedKey,
+    TensorDict,
+    TensorDictBase,
+    unravel_key,
+)
 from tensordict.nn import ProbabilisticTensorDictModule, TensorDictParams
-from tensordict.utils import is_seq_of_nested_key
+from tensordict.utils import _zip_strict, is_seq_of_nested_key
 from torch import nn
-from torchrl.data.tensor_specs import Composite, Unbounded
-from torchrl.envs.transforms.transforms import Transform
+from torchrl.data.tensor_specs import Composite, NonTensor, TensorSpec, Unbounded
+from torchrl.envs.transforms.transforms import TensorDictPrimer, Transform
 from torchrl.envs.transforms.utils import _set_missing_tolerance, _stateless_param
 
 
+def as_nested_tensor(list_of_tensordicts: list[TensorDictBase]) -> TensorDictBase:
+    """Stacks a list of tensordicts into a single tensordict with nested tensors.
+
+    Args:
+        list_of_tensordicts (list[TensorDictBase]): A list of tensordicts to stack.
+
+    Returns:
+        TensorDictBase: A tensordict with nested tensors.
+
+    """
+
+    def _as_nested_tensor(*list_of_tensors):
+        return torch.nested.as_nested_tensor(list_of_tensors, layout=torch.jagged)
+
+    batch_size = list(list_of_tensordicts[0].shape)
+    batch_size.insert(0, len(list_of_tensordicts))
+    return list_of_tensordicts[0].apply(
+        _as_nested_tensor, *list_of_tensordicts[1:], batch_size=batch_size
+    )
+
+
+def as_padded_tensor(
+    list_of_tensordicts: list[[TensorDictBase]], dim=0, stack_dim: int = 0
+) -> TensorDictBase:
+    """Stacks a list of tensordicts into a single tensordict with padded tensors.
+
+    Args:
+        list_of_tensordicts (list[[TensorDictBase]]): A list of tensordicts to stack.
+        dim (int, optional): The dimension along which to pad. Defaults to 0.
+        stack_dim (int, optional): The dimension along which to stack. Defaults to 0.
+
+    Returns:
+        TensorDictBase: A tensordict with padded tensors.
+    """
+
+    def _stack_tensors(*list_of_tensors):
+        if dim < 0:
+            raise ValueError("dim must be >= 0")
+        max_length = max([t.size(dim) for t in list_of_tensors])
+
+        def pad_tensor(tensor):
+            padding_length = max_length - tensor.size(dim)
+            shape = [
+                s if i != dim else padding_length for i, s in enumerate(tensor.shape)
+            ]
+            return torch.cat((tensor.new_zeros(shape), tensor), dim=dim)
+
+        return torch.stack([pad_tensor(t) for t in list_of_tensors], dim=stack_dim)
+
+    batch_size = list(list_of_tensordicts[0].shape)
+    batch_size.insert(dim, len(list_of_tensordicts))
+    result = list_of_tensordicts[0].apply(
+        _stack_tensors, *list_of_tensordicts[1:], batch_size=batch_size
+    )
+    return result
+
+
+class DataLoadingPrimer(TensorDictPrimer):
+    """A primer that loads data from a dataloader and stacks it into a tensordict.
```

Comment:
nit: As a n00b, the term ``stack`` is not as familiar to me (though I know it is used extensively within tensordict). Feel free to ignore this suggestion though

```suggestion
    """A primer that loads data from a dataloader and converts it into a tensordict using ``maybe_dense_stack`` or the provided ``stack_method``.
```